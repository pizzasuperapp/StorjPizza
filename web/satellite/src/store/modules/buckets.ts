// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { Bucket, BucketCursor, BucketPage, BucketsApi } from '@/types/buckets';
import {StoreModule} from "@/types/store";

export const BUCKET_ACTIONS = {
    FETCH: 'setBuckets',
    FETCH_ALL_BUCKET_NAMES: 'getAllBucketNames',
    SET_SEARCH: 'setBucketSearch',
    CLEAR: 'clearBuckets',
};

export const BUCKET_MUTATIONS = {
    SET: 'setBuckets',
    SET_ALL_BUCKET_NAMES: 'setAllBucketNames',
    SET_SEARCH: 'setBucketSearch',
    SET_PAGE: 'setBucketPage',
    CLEAR: 'clearBuckets',
};

const {
    FETCH,
    FETCH_ALL_BUCKET_NAMES,
} = BUCKET_ACTIONS;
const {
    SET,
    SET_ALL_BUCKET_NAMES,
    SET_PAGE,
    SET_SEARCH,
    CLEAR,
} = BUCKET_MUTATIONS;
const bucketPageLimit = 7;
const firstPage = 1;

export class BucketsState {
    public allBucketNames: string[] = [];
    public cursor: BucketCursor = { limit: bucketPageLimit, search: '', page: firstPage };
    public page: BucketPage = { buckets: new Array<Bucket>(), currentPage: 1, pageCount: 1, offset: 0, limit: bucketPageLimit, search: '', totalCount: 0 };
}

interface BucketsContext {
    state: BucketsState
    commit: (string, ...unknown) => void
    rootGetters: {
        selectedProject: {
            id: string
        }
    }
}

/**
 * creates buckets module with all dependencies
 *
 * @param api - buckets api
 */
export function makeBucketsModule(api: BucketsApi): StoreModule<BucketsState, BucketsContext> {
    return {
        state: new BucketsState(),

        mutations: {
            [SET](state: BucketsState, page: BucketPage) {
                state.page = page;
            },
            [SET_ALL_BUCKET_NAMES](state: BucketsState, allBucketNames: string[]) {
                state.allBucketNames = allBucketNames;
            },
            [SET_PAGE](state: BucketsState, page: number) {
                state.cursor.page = page;
            },
            [SET_SEARCH](state: BucketsState, search: string) {
                state.cursor.search = search;
            },
            [CLEAR](state: BucketsState) {
                state.allBucketNames = [];
                state.cursor = new BucketCursor('', bucketPageLimit, firstPage);
                state.page = new BucketPage([], '', bucketPageLimit, 0, 1, 1, 0);
            },
        },
        actions: {
            [FETCH]: async function({commit, rootGetters, state}: BucketsContext, page: number): Promise<BucketPage> {
                const projectID = rootGetters.selectedProject.id;
                const before = new Date();
                state.cursor.page = page;

                commit(SET_PAGE, page);

                const result: BucketPage = await api.get(projectID, before, state.cursor);

                commit(SET, result);

                return result;
            },
            [FETCH_ALL_BUCKET_NAMES]: async function({commit, rootGetters}: BucketsContext): Promise<string[]> {
                const result: string[] = await api.getAllBucketNames(rootGetters.selectedProject.id);

                commit(SET_ALL_BUCKET_NAMES, result);

                return result;
            },
            [BUCKET_ACTIONS.SET_SEARCH]: function({commit}: BucketsContext, search: string) {
                commit(SET_SEARCH, search);
            },
            [BUCKET_ACTIONS.CLEAR]: function({commit}: BucketsContext) {
                commit(CLEAR);
            },
        },
        getters: {
            page: (state: BucketsState): BucketPage => state.page,
            cursor: (state: BucketsState): BucketCursor => state.cursor,
        },
    };
}
