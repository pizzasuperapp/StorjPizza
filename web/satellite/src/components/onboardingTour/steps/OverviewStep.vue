// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="overview-area">
        <h1 class="overview-area__title" aria-roledescription="title">Welcome to Storj {{ titleLabel }}</h1>
        <p class="overview-area__subtitle">Get started using the web browser, or the command line.</p>
        <div class="overview-area__routes">
            <OverviewContainer
                is-web="true"
                title="Start with web browser"
                info="Start uploading files in the browser and instantly see how your data gets distributed over the Storj network around the world."
                button-label="Continue in web ->"
                :on-click="onUploadInBrowserClick"
                :is-disabled="isLoading"
            />
            <OverviewContainer
                title="Start with Uplink CLI"
                info="The Uplink CLI is a command-line interface tool which allows you to upload and download files from the network, manage permissions and share files."
                button-label="Continue in cli ->"
                :on-click="onUplinkCLIClick"
                :is-disabled="isLoading"
            />
        </div>
        <router-link
            class="overview-area__skip-button"
            :to="projectDashboardPath"
        >
            Skip and go directly to dashboard
        </router-link>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import OverviewContainer from '@/components/onboardingTour/steps/common/OverviewContainer.vue';

import { AnalyticsHttpApi } from '@/api/analytics';
import { RouteConfig } from '@/router';
import { AnalyticsEvent } from '@/utils/constants/analyticsEventNames';
import { MetaUtils } from "@/utils/meta";

// @vue/component
@Component({
    components: {
        OverviewContainer,
    },
})
export default class OverviewStep extends Vue {
    public isLoading = false;
    public projectDashboardPath = RouteConfig.ProjectDashboard.path;
    public titleLabel = '';

    private readonly analytics: AnalyticsHttpApi = new AnalyticsHttpApi();

    /**
     * Mounted hook after initial render.
     * Sets correct title label.
     */
    public mounted(): void {
        const partneredSatellites = MetaUtils.getMetaContent('partnered-satellites');
        if (!partneredSatellites) {
            this.titleLabel = 'OSP';
            return;
        }

        const partneredSatellitesJSON = JSON.parse(partneredSatellites);
        const isPartnered = partneredSatellitesJSON.find(el => {
            return el[0] === this.satelliteName;
        })
        if (isPartnered) {
            this.titleLabel = 'DCS';
            return;
        }

        this.titleLabel = 'OSP';
    }

    /**
     * Holds button click logic.
     * Redirects to next step (creating access grant).
     */
    public async onUplinkCLIClick(): Promise<void> {
        if (this.isLoading) return;

        this.isLoading = true;

        await this.analytics.linkEventTriggered(AnalyticsEvent.PATH_SELECTED, 'CLI');
        this.analytics.pageVisit(RouteConfig.OnboardingTour.with(RouteConfig.CLIStep).with(RouteConfig.AGName).path);
        await this.$router.push(RouteConfig.OnboardingTour.with(RouteConfig.CLIStep).with(RouteConfig.AGName).path);

        this.isLoading = false;
    }

    /**
     * Redirects to buckets page.
     */
    public async onUploadInBrowserClick(): Promise<void> {
        if (this.isLoading) return;

        this.isLoading = true;

        await this.analytics.linkEventTriggered(AnalyticsEvent.PATH_SELECTED, 'Continue in Browser');
        this.analytics.pageVisit(RouteConfig.Buckets.path);
        await this.$router.push(RouteConfig.Buckets.path).catch(() => {return; });

        this.isLoading = false;
    }

    private get satelliteName(): string {
        return this.$store.state.appStateModule.satelliteName;
    }
}
</script>

<style scoped lang="scss">
    .overview-area {
        display: flex;
        flex-direction: column;
        align-items: center;
        font-family: 'font_regular', sans-serif;

        &__title {
            font-family: 'font_bold', sans-serif;
            color: #14142b;
            font-size: 32px;
            line-height: 39px;
            margin-bottom: 12.5px;
        }

        &__subtitle {
            font-family: 'font_regular', sans-serif;
            font-weight: 400;
            text-align: center;
            color: #354049;
            font-size: 16px;
            line-height: 21px;
        }

        &__routes {
            margin-top: 35px;
            display: flex;
            align-items: center;
            justify-content: center;
            flex-wrap: wrap;
            column-gap: 38px;
            row-gap: 38px;
        }

        &__skip-button {
            margin-top: 58px;
            color: #b7c1ca;
            cursor: pointer;
            text-decoration: underline !important;

            &:hover {
                text-decoration: underline;
            }
        }
    }
</style>
