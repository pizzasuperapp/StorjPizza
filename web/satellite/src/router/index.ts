// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Router from 'vue-router';

import AccessGrants from '@/components/accessGrants/AccessGrants.vue';
import CreateAccessGrant from '@/components/accessGrants/CreateAccessGrant.vue';
import CLIStep from '@/components/accessGrants/steps/CLIStep.vue';
import CreatePassphraseStep from '@/components/accessGrants/steps/CreatePassphraseStep.vue';
import EnterPassphraseStep from '@/components/accessGrants/steps/EnterPassphraseStep.vue';
import GatewayStep from '@/components/accessGrants/steps/GatewayStep.vue';
import NameStep from '@/components/accessGrants/steps/NameStep.vue';
import PermissionsStep from '@/components/accessGrants/steps/PermissionsStep.vue';
import ResultStep from '@/components/accessGrants/steps/ResultStep.vue';
import AccountArea from '@/components/account/AccountArea.vue';
import AccountBilling from '@/components/account/billing/BillingArea.vue';

// new payment screen routes
import BillingOverview from '@/components/account/billing/billingTabs/Overview.vue';
import BillingPaymentMethods from '@/components/account/billing/billingTabs/PaymentMethods.vue';
import BillingHistory2 from '@/components/account/billing/billingTabs/BillingHistory.vue';
import BillingCoupons from '@/components/account/billing/billingTabs/Coupons.vue';


import DetailedHistory from '@/components/account/billing/depositAndBillingHistory/DetailedHistory.vue';
import AddCouponCode from '@/components/account/billing/coupons/AddCouponCode.vue';
import CreditsHistory from '@/components/account/billing/coupons/CouponArea.vue';
import SettingsArea from '@/components/account/SettingsArea.vue';
import Page404 from '@/components/errors/Page404.vue';
import BucketsView from '@/components/objects/BucketsView.vue';
import EncryptData from '@/components/objects/EncryptData.vue';
import ObjectsArea from '@/components/objects/ObjectsArea.vue';
import UploadFile from '@/components/objects/UploadFile.vue';
import OnboardingTourArea from '@/components/onboardingTour/OnboardingTourArea.vue';
import OnbCLIStep from '@/components/onboardingTour/steps/CLIStep.vue';
import OverviewStep from '@/components/onboardingTour/steps/OverviewStep.vue';
import CreateProject from '@/components/project/CreateProject.vue';
import EditProjectDetails from '@/components/project/EditProjectDetails.vue';
import ProjectDashboard from '@/components/project/ProjectDashboard.vue';
import NewProjectDashboard from "@/components/project/newProjectDashboard/NewProjectDashboard.vue";
import ProjectsList from '@/components/projectsList/ProjectsList.vue';
import ProjectMembersArea from '@/components/team/ProjectMembersArea.vue';
import CLIInstall from "@/components/onboardingTour/steps/cliFlow/CLIInstall.vue";
import APIKey from "@/components/onboardingTour/steps/cliFlow/APIKey.vue";
import CLISetup from "@/components/onboardingTour/steps/cliFlow/CLISetup.vue";
import CreateBucket from "@/components/onboardingTour/steps/cliFlow/CreateBucket.vue";
import UploadObject from "@/components/onboardingTour/steps/cliFlow/UploadObject.vue";
import ListObject from "@/components/onboardingTour/steps/cliFlow/ListObject.vue";
import DownloadObject from "@/components/onboardingTour/steps/cliFlow/DownloadObject.vue";
import ShareObject from "@/components/onboardingTour/steps/cliFlow/ShareObject.vue";
import RegistrationSuccess from "@/components/common/RegistrationSuccess.vue";
import SuccessScreen from "@/components/onboardingTour/steps/cliFlow/SuccessScreen.vue";
import AGName from "@/components/onboardingTour/steps/cliFlow/AGName.vue";
import AGPermissions from "@/components/onboardingTour/steps/cliFlow/AGPermissions.vue";
import BucketCreation from "@/components/objects/BucketCreation.vue";

import { NavigationLink } from '@/types/navigation';
import { MetaUtils } from "@/utils/meta";
import BucketDetails from "@/components/objects/BucketDetails.vue";

const ActivateAccount = () => import('@/views/ActivateAccount.vue');
const AuthorizeArea = () => import('@/views/AuthorizeArea.vue');
const DashboardArea = () => import('@/views/DashboardArea.vue');
const ForgotPassword = () => import('@/views/ForgotPassword.vue');
const LoginArea = () => import('@/views/LoginArea.vue');
const RegisterArea = () => import('@/views/RegisterArea.vue');
const ResetPassword = () => import('@/views/ResetPassword.vue');

Vue.use(Router);

/**
 * RouteConfig contains information about all routes and subroutes
 */
export abstract class RouteConfig {
    // root paths
    public static Root = new NavigationLink('/', 'Root');
    public static Login = new NavigationLink('/login', 'Login');
    public static Register = new NavigationLink('/signup', 'Register');
    public static RegisterSuccess = new NavigationLink('/signup-success', 'RegisterSuccess');
    public static Activate = new NavigationLink('/activate', 'Activate');
    public static ForgotPassword = new NavigationLink('/forgot-password', 'Forgot Password');
    public static ResetPassword = new NavigationLink('/password-recovery', 'Reset Password');
    public static Authorize = new NavigationLink('/oauth/v2/authorize', 'Authorize')
    public static Account = new NavigationLink('/account', 'Account');
    public static ProjectDashboard = new NavigationLink('/project-dashboard', 'Project Overview');
    public static NewProjectDashboard = new NavigationLink('/new-project-dashboard', 'Project Overview ');
    public static Users = new NavigationLink('/project-members', 'Users');
    public static OnboardingTour = new NavigationLink('/onboarding-tour', 'Onboarding Tour');
    public static CreateProject = new NavigationLink('/create-project', 'Create Project');
    public static EditProjectDetails = new NavigationLink('/edit-project-details', 'Edit Project Details');
    public static AccessGrants = new NavigationLink('/access-grants', 'Access');
    public static ProjectsList = new NavigationLink('/projects', 'Projects');
    public static Buckets = new NavigationLink('/buckets', 'Buckets');

    // account child paths
    public static Settings = new NavigationLink('settings', 'Settings');
    public static Billing = new NavigationLink('billing', 'Billing');
    public static AddCouponCode = new NavigationLink('add-coupon', 'Get Free Credits');
    public static BillingHistory = new NavigationLink('billing-history', 'Billing History');
    public static BillingOverview = new NavigationLink('overview', 'Overview');
    public static BillingPaymentMethods = new NavigationLink('payment-methods', 'Payment Methods');
    public static BillingHistory2 = new NavigationLink('billing-history2', 'Billing History 2');
    public static BillingCoupons = new NavigationLink('coupons', 'Coupons');
    public static DepositHistory = new NavigationLink('deposit-history', 'Deposit History');
    public static CreditsHistory = new NavigationLink('credits-history', 'Credits History');

    // access grant child paths
    public static CreateAccessGrant = new NavigationLink('create-grant', 'Create Access Grant');
    public static NameStep = new NavigationLink('name', 'Name Access Grant');
    public static PermissionsStep = new NavigationLink('permissions', 'Access Grant Permissions');
    public static CreatePassphraseStep = new NavigationLink('create-passphrase', 'Access Grant Create Passphrase');
    public static EnterPassphraseStep = new NavigationLink('enter-passphrase', 'Access Grant Enter Passphrase');
    public static ResultStep = new NavigationLink('result', 'Access Grant Result');
    public static GatewayStep = new NavigationLink('gateway', 'Access Grant Gateway');
    public static CLIStep = new NavigationLink('cli', 'Access Grant In CLI');

    // onboarding tour child paths
    public static OverviewStep = new NavigationLink('overview', 'Onboarding Overview');
    public static OnbCLIStep = new NavigationLink('cli', 'Onboarding CLI');
    public static AGName = new NavigationLink('ag-name', 'Onboarding AG Name');
    public static AGPermissions = new NavigationLink('ag-permissions', 'Onboarding AG Permissions');
    public static APIKey = new NavigationLink('api-key', 'Onboarding API Key');
    public static CLIInstall = new NavigationLink('cli-install', 'Onboarding CLI Install');
    public static CLISetup = new NavigationLink('cli-setup', 'Onboarding CLI Setup');
    public static CreateBucket = new NavigationLink('create-bucket', 'Onboarding Create Bucket');
    public static UploadObject = new NavigationLink('upload-object', 'Onboarding Upload Object');
    public static ListObject = new NavigationLink('list-object', 'Onboarding List Object');
    public static DownloadObject = new NavigationLink('download-object', 'Onboarding Download Object');
    public static ShareObject = new NavigationLink('share-object', 'Onboarding Share Object');
    public static SuccessScreen = new NavigationLink('success', 'Onboarding Success Screen');

    // objects child paths.
    public static EncryptData = new NavigationLink('encrypt-data', 'Objects Encrypt Data');
    public static BucketsManagement = new NavigationLink('management', 'Buckets Management');
    public static BucketsDetails = new NavigationLink('details', 'Bucket Details');
    public static UploadFile = new NavigationLink('upload/', 'Objects Upload');
    public static UploadFileChildren = new NavigationLink('*', 'Objects Upload Children');
    public static BucketCreation = new NavigationLink('creation', 'Bucket Creation')
}

const isNewProjectDashboard = MetaUtils.getMetaContent('new-project-dashboard') === 'true';
if (isNewProjectDashboard) {
    RouteConfig.ProjectDashboard = RouteConfig.NewProjectDashboard
}

export const notProjectRelatedRoutes = [
    RouteConfig.Login.name,
    RouteConfig.Register.name,
    RouteConfig.RegisterSuccess.name,
    RouteConfig.Activate.name,
    RouteConfig.ForgotPassword.name,
    RouteConfig.ResetPassword.name,
    RouteConfig.Authorize.name,
    RouteConfig.Billing.name,
    RouteConfig.BillingHistory.name,
    RouteConfig.BillingOverview.name,
    RouteConfig.BillingPaymentMethods.name,
    RouteConfig.BillingHistory2.name,
    RouteConfig.BillingCoupons.name,
    RouteConfig.DepositHistory.name,
    RouteConfig.CreditsHistory.name,
    RouteConfig.Settings.name,
];

export const router = new Router({
    mode: 'history',
    routes: [
        {
            path: RouteConfig.Login.path,
            name: RouteConfig.Login.name,
            component: LoginArea,
        },
        {
            path: RouteConfig.Register.path,
            name: RouteConfig.Register.name,
            component: RegisterArea,
        },
        {
            path: RouteConfig.RegisterSuccess.path,
            name: RouteConfig.RegisterSuccess.name,
            component: RegistrationSuccess,
        },
        {
            path: RouteConfig.Activate.path,
            name: RouteConfig.Activate.name,
            component: ActivateAccount,
        },
        {
            path: RouteConfig.ForgotPassword.path,
            name: RouteConfig.ForgotPassword.name,
            component: ForgotPassword,
        },
        {
            path: RouteConfig.ResetPassword.path,
            name: RouteConfig.ResetPassword.name,
            component: ResetPassword,
        },
        {
            path: RouteConfig.Authorize.path,
            name: RouteConfig.Authorize.name,
            component: AuthorizeArea,
        },
        {
            path: RouteConfig.Root.path,
            meta: {
                requiresAuth: true,
            },
            component: DashboardArea,
            children: [
                {
                    path: RouteConfig.Root.path,
                    name: 'default',
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Account.path,
                    name: RouteConfig.Account.name,
                    component: AccountArea,
                    children: [
                        {
                            path: RouteConfig.Settings.path,
                            name: RouteConfig.Settings.name,
                            component: SettingsArea,
                        },
                        {
                            path: RouteConfig.Billing.path,
                            name: RouteConfig.Billing.name,
                            component: AccountBilling,
                            children: [
                                {
                                    path: RouteConfig.AddCouponCode.path,
                                    name: RouteConfig.AddCouponCode.name,
                                    component: AddCouponCode,
                                },
                                {
                                    path: RouteConfig.BillingOverview.path,
                                    name: RouteConfig.BillingOverview.name,
                                    component: BillingOverview,
                                },
                                {
                                    path: RouteConfig.BillingPaymentMethods.path,
                                    name: RouteConfig.BillingPaymentMethods.name,
                                    component: BillingPaymentMethods,
                                },
                                {
                                    path: RouteConfig.BillingHistory2.path,
                                    name: RouteConfig.BillingHistory2.name,
                                    component: BillingHistory2,
                                },
                                {
                                    path: RouteConfig.BillingCoupons.path,
                                    name: RouteConfig.BillingCoupons.name,
                                    component: BillingCoupons,
                                },
                            ],
                        },
                        {
                            path: RouteConfig.BillingHistory.path,
                            name: RouteConfig.BillingHistory.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.DepositHistory.path,
                            name: RouteConfig.DepositHistory.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.CreditsHistory.path,
                            name: RouteConfig.CreditsHistory.name,
                            component: CreditsHistory,
                        },
                    ],
                },
                {
                    path: RouteConfig.NewProjectDashboard.path,
                    name: RouteConfig.NewProjectDashboard.name,
                    component: NewProjectDashboard,
                },
                {
                    path: RouteConfig.ProjectDashboard.path,
                    name: RouteConfig.ProjectDashboard.name,
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Users.path,
                    name: RouteConfig.Users.name,
                    component: ProjectMembersArea,
                },
                {
                    path: RouteConfig.OnboardingTour.path,
                    name: RouteConfig.OnboardingTour.name,
                    component: OnboardingTourArea,
                    children: [
                        {
                            path: RouteConfig.OverviewStep.path,
                            name: RouteConfig.OverviewStep.name,
                            component: OverviewStep,
                        },
                        {
                            path: RouteConfig.OnbCLIStep.path,
                            name: RouteConfig.OnbCLIStep.name,
                            component: OnbCLIStep,
                            children: [
                                {
                                    path: RouteConfig.AGName.path,
                                    name: RouteConfig.AGName.name,
                                    component: AGName,
                                },
                                {
                                    path: RouteConfig.AGPermissions.path,
                                    name: RouteConfig.AGPermissions.name,
                                    component: AGPermissions,
                                },
                                {
                                    path: RouteConfig.APIKey.path,
                                    name: RouteConfig.APIKey.name,
                                    component: APIKey,
                                },
                                {
                                    path: RouteConfig.CLIInstall.path,
                                    name: RouteConfig.CLIInstall.name,
                                    component: CLIInstall,
                                },
                                {
                                    path: RouteConfig.CLISetup.path,
                                    name: RouteConfig.CLISetup.name,
                                    component: CLISetup,
                                },
                                {
                                    path: RouteConfig.CreateBucket.path,
                                    name: RouteConfig.CreateBucket.name,
                                    component: CreateBucket,
                                },
                                {
                                    path: RouteConfig.UploadObject.path,
                                    name: RouteConfig.UploadObject.name,
                                    component: UploadObject,
                                },
                                {
                                    path: RouteConfig.ListObject.path,
                                    name: RouteConfig.ListObject.name,
                                    component: ListObject,
                                },
                                {
                                    path: RouteConfig.DownloadObject.path,
                                    name: RouteConfig.DownloadObject.name,
                                    component: DownloadObject,
                                },
                                {
                                    path: RouteConfig.ShareObject.path,
                                    name: RouteConfig.ShareObject.name,
                                    component: ShareObject,
                                },
                                {
                                    path: RouteConfig.SuccessScreen.path,
                                    name: RouteConfig.SuccessScreen.name,
                                    component: SuccessScreen,
                                },
                            ],
                        },
                    ],
                },
                {
                    path: RouteConfig.CreateProject.path,
                    name: RouteConfig.CreateProject.name,
                    component: CreateProject,
                },
                {
                    path: RouteConfig.EditProjectDetails.path,
                    name: RouteConfig.EditProjectDetails.name,
                    component: EditProjectDetails,
                },
                {
                    path: RouteConfig.AccessGrants.path,
                    name: RouteConfig.AccessGrants.name,
                    component: AccessGrants,
                    children: [
                        {
                            path: RouteConfig.CreateAccessGrant.path,
                            name: RouteConfig.CreateAccessGrant.name,
                            component: CreateAccessGrant,
                            children: [
                                {
                                    path: RouteConfig.NameStep.path,
                                    name: RouteConfig.NameStep.name,
                                    component: NameStep,
                                },
                                {
                                    path: RouteConfig.PermissionsStep.path,
                                    name: RouteConfig.PermissionsStep.name,
                                    component: PermissionsStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.CreatePassphraseStep.path,
                                    name: RouteConfig.CreatePassphraseStep.name,
                                    component: CreatePassphraseStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.EnterPassphraseStep.path,
                                    name: RouteConfig.EnterPassphraseStep.name,
                                    component: EnterPassphraseStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.ResultStep.path,
                                    name: RouteConfig.ResultStep.name,
                                    component: ResultStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.GatewayStep.path,
                                    name: RouteConfig.GatewayStep.name,
                                    component: GatewayStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.CLIStep.path,
                                    name: RouteConfig.CLIStep.name,
                                    component: CLIStep,
                                    props: true,
                                },
                            ],
                        },
                    ],
                },
                {
                    path: RouteConfig.ProjectsList.path,
                    name: RouteConfig.ProjectsList.name,
                    component: ProjectsList,
                },
                {
                    path: RouteConfig.Buckets.path,
                    name: RouteConfig.Buckets.name,
                    component: ObjectsArea,
                    children: [
                        {
                            path: RouteConfig.EncryptData.path,
                            name: RouteConfig.EncryptData.name,
                            component: EncryptData,
                        },
                        {
                            path: RouteConfig.BucketsManagement.path,
                            name: RouteConfig.BucketsManagement.name,
                            component: BucketsView,
                            props: true,
                        },
                        {
                            path: RouteConfig.BucketsDetails.path,
                            name: RouteConfig.BucketsDetails.name,
                            component: BucketDetails,
                            props: true,
                        },
                        {
                            path: RouteConfig.UploadFile.path,
                            name: RouteConfig.UploadFile.name,
                            component: UploadFile,
                            children: [
                                {
                                    path: RouteConfig.UploadFileChildren.path,
                                    name: RouteConfig.UploadFileChildren.name,
                                    component: UploadFile,
                                },
                            ],
                        },
                        {
                            path: RouteConfig.BucketCreation.path,
                            name: RouteConfig.BucketCreation.name,
                            component: BucketCreation,
                        },
                    ],
                },
            ],
        },
        {
            path: '*',
            name: '404',
            component: Page404,
        },
    ],
});
