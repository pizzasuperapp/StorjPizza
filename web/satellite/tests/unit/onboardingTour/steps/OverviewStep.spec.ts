// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import OverviewStep from '@/components/onboardingTour/steps/OverviewStep.vue';

import { router } from '@/router';
import { appStateModule } from "@/store/modules/appState";
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({ modules: { appStateModule }});

describe('OverviewStep.vue', (): void => {
    it('renders correctly', (): void => {
        const wrapper = mount(OverviewStep, {
            localVue,
            router,
            store,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
