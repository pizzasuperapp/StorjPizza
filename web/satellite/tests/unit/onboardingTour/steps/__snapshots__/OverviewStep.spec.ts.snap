// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewStep.vue renders correctly 1`] = `
<div class="overview-area">
  <h1 aria-roledescription="title" class="overview-area__title">Welcome to Storj </h1>
  <p class="overview-area__subtitle">Get started using the web browser, or the command line.</p>
  <div class="overview-area__routes">
    <div class="overview-container"><svg alt="web" class="overview-container__img"></svg>
      <p aria-roledescription="server-side-encryption-title" class="overview-container__enc">Server-side encrypted</p>
      <h2 class="overview-container__title">Start with web browser</h2>
      <p class="overview-container__info">Start uploading files in the browser and instantly see how your data gets distributed over the Storj network around the world.</p>
      <div class="container" style="width: 240px; height: 48px; border-radius: 8px; font-size: 16px;">
        <!---->
        <!---->
        <!----> <span class="label">Continue in web -&gt;</span>
      </div>
      <p class="overview-container__encryption-container">
        By using the web browser you are opting in to
        <a href="https://docs.storj.io/concepts/encryption-key/design-decision-server-side-encryption" target="_blank" rel="noopener noreferrer" aria-roledescription="server-side-encryption-link" class="overview-container__encryption-container__link">server-side encryption</a>.
      </p>
    </div>
    <div class="overview-container"><svg alt="cli" class="overview-container__img"></svg>
      <p aria-roledescription="end-to-end-encryption-title" class="overview-container__enc">End-to-end encrypted</p>
      <h2 class="overview-container__title">Start with Uplink CLI</h2>
      <p class="overview-container__info">The Uplink CLI is a command-line interface tool which allows you to upload and download files from the network, manage permissions and share files.</p>
      <div class="container" style="width: 240px; height: 48px; border-radius: 8px; font-size: 16px;">
        <!---->
        <!---->
        <!----> <span class="label">Continue in cli -&gt;</span>
      </div>
      <p class="overview-container__encryption-container">
        The Uplink CLI uses
        <a href="https://docs.storj.io/concepts/encryption-key/design-decision-end-to-end-encryption" target="_blank" rel="noopener noreferrer" aria-roledescription="end-to-end-encryption-link" class="overview-container__encryption-container__link">end-to-end encryption</a> for object data, metadata and path data.
      </p>
    </div>
  </div> <a href="/project-dashboard" class="overview-area__skip-button">
    Skip and go directly to dashboard
  </a>
</div>
`;
